bittensor/_subtensor/extrinsics/registration.py

```diff
@@ -178,4 +178,97 @@ def register_extrinsic (
+    subtensor: 'bittensor.Subtensor',
+    wallet: 'bittensor.Wallet',
+    netuid: int,
+    wait_for_inclusion: bool = False,
+    wait_for_finalization: bool = True,
+    prompt: bool = False
+) -> bool:
+    r""" Registers the wallet to chain by burning TAO.
+    Args:
+        wallet (bittensor.wallet):
+            bittensor wallet object.
+        netuid (int):
+            The netuid of the subnet to register on.
+        wait_for_inclusion (bool):
+            If set, waits for the extrinsic to enter a block before returning true, 
+            or returns false if the extrinsic fails to enter the block within the timeout.   
+        wait_for_finalization (bool):
+            If set, waits for the extrinsic to be finalized on the chain before returning true,
+            or returns false if the extrinsic fails to be finalized within the timeout.
+        prompt (bool):
+            If true, the call waits for confirmation from the user before proceeding.
+    Returns:
+        success (bool):
+            flag is true if extrinsic was finalized or uncluded in the block. 
+            If we did not wait for finalization / inclusion, the response is true.
+    """
+    if not subtensor.subnet_exists( netuid ):
+        bittensor.__console__.print(":cross_mark: [red]Failed[/red]: error: [bold white]subnet:{}[/bold white] does not exist.".format(netuid))
+        return False
+
+    with bittensor.__console__.status(f":satellite: Checking Account on [bold]subnet:{netuid}[/bold]..."):
+        neuron = subtensor.get_neuron_for_pubkey_and_subnet( wallet.hotkey.ss58_address, netuid = netuid )
+
+        old_balance = subtensor.get_balance( wallet.coldkeypub.ss58_address )
+
+        burn_amount = subtensor.burn( netuid = netuid )
+        if not neuron.is_null:
+            bittensor.__console__.print(
+            ':white_heavy_check_mark: [green]Already Registered[/green]:\n'\
+            'uid: [bold white]{}[/bold white]\n' \
+            'netuid: [bold white]{}[/bold white]\n' \
+            'hotkey: [bold white]{}[/bold white]\n' \
+            'coldkey: [bold white]{}[/bold white]' 
+            .format(neuron.uid, neuron.netuid, neuron.hotkey, neuron.coldkey))
+            return True
+        
+    if prompt:
+        # Prompt user for confirmation.
+        if not Confirm.ask( f"Burn {burn_amount} to register on subnet:{netuid}?" ):
+            return False
+
+    with bittensor.__console__.status(":satellite: Burning TAO for Registration..."):
+       with subtensor.substrate as substrate:
+            # create extrinsic call
+            call = substrate.compose_call( 
+                call_module='SubtensorModule',  
+                call_function='burned_register', 
+                call_params={ 
+                    'netuid': netuid,
+                    'hotkey': wallet.hotkey.ss58_address
+                } 
+            )
+            extrinsic = substrate.create_signed_extrinsic( call = call, keypair = wallet.hotkey )
```

Comment:
should we use coldkey to sign instead? cause burn reg burns tao (stake)

Reply:
Oops. fixed